### ISSUE TITLE
Would it be possible to not use FTTableMorph>>highlightRowIndexes:?

### ISSUE THREAD

**Issue by jecisc** [pharo-ide/Calypso#439]
Hi,

I am working on FastTable and I see that when I worked on the FTSearchFunction I added the concept of highlighted rows. But now that I am more experienced I see that this variable and all the methods related are useless and I can just implement the search function using #selectIndexes: instead. 

So I wanted to remove the variable and the related methods to reduce the memory consumption of FT and simplifying the code. Simplifications that will help me to add other features. 

While doing the refactoring I see that one of the highlighting methods is used by Calypso. This is only one call. It is in ClyHighlightingSelection.

I am wondering if it would be possible to implement this method in a different way to be able to clean this code that is kind of a duplication of the selection mechanism? I don't have much time to look at it now :(

**Comment by dionisiydk** [comment]
Calypso uses it to highlight protocols of selected method. How you are
going to support this feature in different way?

2 февр. 2019 г. 19:51 пользователь "CyrilFerlicot" <notifications@github.com>
написал:

Hi,

I am working on FastTable and I see that when I worked on the
FTSearchFunction I added the concept of highlighted rows. But now that I am
more experienced I see that this variable and all the methods related are
useless and I can just implement the search function using #selectIndexes:
instead.

So I wanted to remove the variable and the related methods to reduce the
memory consumption of FT and simplifying the code. Simplifications that
will help me to add other features.

While doing the refactoring I see that one of the highlighting methods is
used by Calypso. This is only one call. It is in ClyHighlightingSelection.

I am wondering if it would be possible to implement this method in a
different way to be able to clean this code that is kind of a duplication
of the selection mechanism? I don't have much time to look at it now :(

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/pharo-ide/Calypso/issues/439>, or mute the thread
